lib/engine/action/hex_token.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Action
+    class HexToken < Base
+      attr_reader :hex, :token
+
+      def initialize(entity, hex:, token_type: nil)
+        super(entity)
+        @hex = hex
+        # token may be nil because when you upgrade someone's 00
+        # and place their token, you pretend to be them and you may not have a token
```

Comment:
sounds scary can you explain

Reply:
This comes from `Engine::Action::PlaceToken`. Git blame shows it relates to 1836jr
```
      def initialize(entity, city:, slot:, token_type: nil)
        super(entity)
        @city = city
        @slot = slot
        # token may be nil because when you upgrade someone's 00
        # and place their token, you pretend to be them and you may not have a token
        @token = @entity.find_token_by_type(token_type&.to_sym)
      end
```

Reply:
there's no reason to copy the comment

Reply:
I removed the comment, thanks!